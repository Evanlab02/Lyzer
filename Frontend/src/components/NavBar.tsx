import { Menu, Moon, Sun } from "lucide-react";
import { useTheme } from "../hooks/useTheme";
import "../styles/navbar.scss";

type NavBarProps = {
	onMenuClick: (event: React.MouseEvent<SVGElement>) => void;
}

export default function NavBar({ onMenuClick}: NavBarProps) {
	const {isDarkMode, toggleTheme} = useTheme();

	return (
		<div className="navbar">
			<Menu 
				color={isDarkMode ? "white" : "black"} 
				size={28} onClick={onMenuClick} 
				className="icon menu"
				id="navbar-menu"
			/>
			<div>
				â€‹ðŸ‡±â€‹â€‹ðŸ‡¾â€‹â€‹ðŸ‡¿â€‹â€‹ðŸ‡ªâ€‹â€‹ðŸ‡·â€‹
			</div>
			{
				isDarkMode ? (
					<Moon
						id="theme-toggle-dark"
						className="icon theme-toggle"
						color="white"
						size={28}
						onClick={() => toggleTheme()}
					/>
				) : (
					<Sun
						id="theme-toggle-light"
						className="icon theme-toggle"
						color="black"
						size={28}
						onClick={() => toggleTheme()}
					/>
				)
			}
		</div>
	);
}