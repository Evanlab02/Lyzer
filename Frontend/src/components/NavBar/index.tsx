import { Menu, Moon, Sun } from "lucide-react";
import { useTheme } from "../../hooks/useTheme";
import "./styles/index.scss";

interface NavBarProps {
	testId?: string;
	menuToggleTestId?: string;
	themeToggleTestId?: string;
	onMenuClick: (event: React.MouseEvent<SVGElement>) => void;
}

export default function NavBar(props: Readonly<NavBarProps>) {
	const {
		testId,
		menuToggleTestId,
		themeToggleTestId,
		onMenuClick
	} = props;
	
	const {
		isDarkMode, 
		toggleTheme
	} = useTheme();

	return (
		<div className="navbar" data-testid={testId}>
			<Menu 
				color={isDarkMode ? "white" : "black"} 
				size={28} 
				onClick={onMenuClick} 
				className="icon menu"
				id="navbar-menu"
				data-testid={menuToggleTestId}
			/>
			<div>L Y Z E R</div>
			{
				isDarkMode ? 
					<Moon
						id="theme-toggle-dark"
						className="icon theme-toggle"
						color="white"
						size={28}
						data-testid={themeToggleTestId}
						onClick={() => { toggleTheme(); }}
					/>
					:
					<Sun
						id="theme-toggle-light"
						className="icon theme-toggle"
						color="black"
						size={28}
						data-testid={themeToggleTestId}
						onClick={() => { toggleTheme(); }}
					/>
				
			}
		</div>
	);
}