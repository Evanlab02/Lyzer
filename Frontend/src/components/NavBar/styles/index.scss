.navbar {
	width: 100vw; 
	height: 4em; 
	border-bottom: 1px solid var(--secondary-colour); 
	display: flex;
	justify-content: space-between;
	align-items: center;
	align-content: space-between;
	box-sizing: border-box; 
	margin: 0; 
	padding: 0; 

	.icon {
		&.menu {
			margin-left: 20px;
		}

		&.theme-toggle {
			margin-right: 20px;
		}
	}
}