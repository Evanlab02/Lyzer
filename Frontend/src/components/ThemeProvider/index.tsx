import { useEffect, useState } from "react";
import { ThemeContext } from "../../hooks/useTheme";

export const ThemeProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
	const defaultDark = window.matchMedia("(prefers-color-scheme: dark)").matches;

	const [isDarkMode, setDarkMode] = useState(
		localStorage.getItem("isDarkMode") !== null
			? localStorage.getItem("isDarkMode") === "true"
			: defaultDark
	);

	const toggleTheme = () => {
		setDarkMode((prev) => !prev);
	};

	useEffect(() => {
		const theme = isDarkMode ? "dark" : "light";
		localStorage.setItem("isDarkMode", theme === "dark" ? "true" : "false");
		document.documentElement.setAttribute("data-theme", theme);
	}, [isDarkMode]);

	return (
		<ThemeContext.Provider value={{ isDarkMode, toggleTheme }}>
			{children}
		</ThemeContext.Provider>
	);
};