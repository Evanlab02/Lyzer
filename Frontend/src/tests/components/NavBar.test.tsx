import { vi, expect, it } from "vitest";
import { fireEvent, render } from "@testing-library/react";
import ThemeProvider from "../../components/ThemeProvider";
import NavBar from "../../components/NavBar";

it("Navbar renders correctly", () => {
	const { getByTestId } = render(
		<ThemeProvider>
			<NavBar
				testId="lyzer-nav-bar"
				onMenuClick={vi.fn()}
			/>
		</ThemeProvider>
	);

	const navBar = getByTestId("lyzer-nav-bar");
	expect(navBar).toMatchSnapshot();
});

it("Navbar onMenuClick is called", () => {
	const onMenuClick = vi.fn();
    
	const { getByTestId } = render(
		<ThemeProvider>
			<NavBar
				testId="lyzer-nav-bar"
				menuToggleTestId="lyzer-nav-bar-menu-icon"
				onMenuClick={onMenuClick}
			/>
		</ThemeProvider>
	);

	const menuIcon = getByTestId("lyzer-nav-bar-menu-icon");
	fireEvent.click(menuIcon);

	expect(onMenuClick).toHaveBeenCalledTimes(1);
});
