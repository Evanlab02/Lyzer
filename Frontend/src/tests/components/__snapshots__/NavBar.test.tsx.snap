// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar renders correctly 1`] = `
<div
  class="navbar"
  data-testid="lyzer-nav-bar"
>
  <svg
    class="lucide lucide-menu icon menu"
    fill="none"
    height="28"
    id="navbar-menu"
    stroke="black"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="28"
    xmlns="http://www.w3.org/2000/svg"
  >
    <line
      x1="4"
      x2="20"
      y1="12"
      y2="12"
    />
    <line
      x1="4"
      x2="20"
      y1="6"
      y2="6"
    />
    <line
      x1="4"
      x2="20"
      y1="18"
      y2="18"
    />
  </svg>
  <div>
    L Y Z E R
  </div>
  <svg
    class="lucide lucide-sun icon theme-toggle"
    fill="none"
    height="28"
    id="theme-toggle-light"
    stroke="black"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="28"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="12"
      cy="12"
      r="4"
    />
    <path
      d="M12 2v2"
    />
    <path
      d="M12 20v2"
    />
    <path
      d="m4.93 4.93 1.41 1.41"
    />
    <path
      d="m17.66 17.66 1.41 1.41"
    />
    <path
      d="M2 12h2"
    />
    <path
      d="M20 12h2"
    />
    <path
      d="m6.34 17.66-1.41 1.41"
    />
    <path
      d="m19.07 4.93-1.41 1.41"
    />
  </svg>
</div>
`;
